.ci/e2e-expected/ddl-dml-batching.txt

```diff
@@ -1 +1 @@
-ERROR: FAILED_PRECONDITION: Executing updates is not allowed for DDL batches. "CREATE TABLE testtable (id integer PRIMARY KEY, data integer); INSERT INTO testtable (id, data) VALUES (1, 2); INSERT INTO testtable (id, data) VALUES (3, 4)"
+INSERT 0 1
```

Comment:
According to psql documentation, psql only prints the result of the last SQL command in the string. 
"Each SQL command string passed to -c is sent to the server as a single request. Because of this, the server executes it as a single transaction even if the string contains multiple SQL commands, unless there are explicit BEGIN/COMMIT commands included in the string to divide it into multiple transactions. (See [Section 53.2.2.1](https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-MULTI-STATEMENT) for more details about how the server handles multi-query strings.) Also, psql only prints the result of the last SQL command in the string. This is different from the behavior when the same string is read from a file or fed to psql's standard input, because then psql sends each SQL command separately." [(link)](https://www.postgresql.org/docs/current/app-psql.html)

I would also like to have a further discussion on this because I notice that in our design doc, the test cases have outputs for each statement in the batch and I would like to know how you get them.

Reply:
> According to psql documentation, psql only prints the result of the last SQL command in the string. "Each SQL command string passed to -c is sent to the server as a single request. Because of this, the server executes it as a single transaction even if the string contains multiple SQL commands, unless there are explicit BEGIN/COMMIT commands included in the string to divide it into multiple transactions. (See [Section 53.2.2.1](https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-MULTI-STATEMENT) for more details about how the server handles multi-query strings.) Also, psql only prints the result of the last SQL command in the string. This is different from the behavior when the same string is read from a file or fed to psql's standard input, because then psql sends each SQL command separately." [(link)](https://www.postgresql.org/docs/current/app-psql.html)
> 
> I would also like to have a further discussion on this because I notice that in our design doc, the test cases have outputs for each statement in the batch and I would like to know how you get them.

The trick to get `psql` to print out the response for each command is to send the commands to the standard input of `psql`. For this you have to options:
1. Just open `psql` in interactive mode and enter the commands directly into `psql`:

```shell
loite@penguin:~$ psql -h localhost -d knut-test-db
Password for user loite: 
psql (14.1 (Debian 14.1-1.pgdg110+1))
SSL connection (protocol: TLSv1.3, cipher: TLS_AES_256_GCM_SHA384, bits: 256, compression: off)
Type "help" for help.

knut-test-db=# begin; insert into t values (3); insert into t values (4); commit;
BEGIN
INSERT 0 1
INSERT 0 1
COMMIT
knut-test-db=# 
```

2. Use echo to input a batch of commands into the standard input of `psql`:

```shell
loite@penguin:~$ echo "begin; insert into t values (1); insert into t values (2); commit;" | psql -h localhost -d knut-test-db
Password for user loite: 
BEGIN
INSERT 0 1
INSERT 0 1
COMMIT
loite@penguin:~$ 
```

If you repeat the above command to force errors:

```shell
loite@penguin:~$ echo "begin; insert into t values (1); insert into t values (2); commit;" | psql -h localhost -d knut-test-db
BEGIN
ERROR:  duplicate key value violates unique constraint "t_pkey"
DETAIL:  Key (v)=(1) already exists.
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK
loite@penguin:~$ 
```

Note: You can make `psql` stop asking for a password by setting the environment variable PGPASSWORD to the password it should use before running `psql`.

Reply:
Note: If our current `psql` test setup is created in such a way that it only outputs the last command tag (that is; the commands are fed into `psql` using the `-c` flag), then we should leave that as it is for now, and change it in a separate PR.

Reply:
Okay, will leave it as it is.